packages/format-library/src/link/index.js

```diff
@@ -39,6 +40,42 @@ function Edit( {
 	contentRef,
 } ) {
 	const [ addingLink, setAddingLink ] = useState( false );
+	const [ clickedLink, setClickedLink ] = useState( false );
```

Comment:
Instead of managing a `clickedLink` state, can we set the state of the thing we want to do? I.e. - display the link preview.

Reply:
I don't see how we could do that. The `Popover` is within the React tree. Therefore we must setState to trigger a re-render of the component with the new state.

Reply:
Can we use `setAddingLink( true )` instead of adding `clickedLink` state?

Reply:
Yeh try that. Might not need two states.

Reply:
Done :) https://github.com/WordPress/gutenberg/pull/57986/commits/b322bbd5f442672969f7b988ffa32966d0393845